rust/scannerctl/src/interpret/mod.rs

```diff
@@ -193,8 +224,15 @@ pub fn run(
         }
         (Db::InMemory, Some(path)) => {
             let storage = DefaultDispatcher::new(true);
-            let builder = RunBuilder::default().loader(create_fp_loader(&storage, path)?);
-            builder.storage(storage).build().run(script)
+            
+            if let Some(_jsonfile) = path.extension() {
+                let builder = RunBuilder::default().loader(create_loader_from_feed_json(&storage, path)?);
+                builder.storage(storage).build().run(script)
+            
+            } else {
+                let builder = RunBuilder::default().loader(create_fp_loader(&storage, path)?);
+                builder.storage(storage).build().run(script)
+            }
```

Comment:
We can minimize the size of the if expression here:

```rust

            let loader = if let Some(_jsonfile) = path.extension() {
                create_loader_from_feed_json(&storage, path)?
            } else {
                create_fp_loader(&storage, path)?
            };
            let builder = RunBuilder::default().loader(loader);
            builder.storage(storage).build().run(script)
```

Also, this doesn't actually check if the file is a json, right? So maybe we should do something like

```rust
            let is_json = path.extension().map(|ext| ext == "json").unwrap_or(false);
            let loader = if is_json {
                create_loader_from_feed_json(&storage, path)?
            } else {
                create_fp_loader(&storage, path)?
            };
```